apps/web/src/app/[locale]/aot-2023/_components/card-grid.tsx

```diff
@@ -0,0 +1,56 @@
+import { prisma } from '@repo/db';
+import { ChallengeCard } from './challenge-card';
+
+const slugs = [
+  'hello-world',
+  'pick',
+  'awaited',
+  'readonly',
+  'exclude',
+  'parameters',
+  'omit',
+  'pop',
+  'trim-left',
+  'get-return-type',
+  'permutation',
+  'currying-1',
+];
+
+export async function CardGrid() {
+  const challenges = await getChallenges();
+  const challengesToReveal = revealItems(challenges);
+
+  return (
+    <div className="container">
+      <section className="w-[calc(100% + 8rem)] grid grid-cols-1 gap-4 sm:px-8 md:-mx-16 md:grid-cols-2 md:px-0 lg:mx-0 lg:w-full xl:grid-cols-3 2xl:gap-8">
+        {challengesToReveal?.map((challenge, i) => {
+          return <ChallengeCard key={i} challenge={challenge} />;
+        })}
+      </section>
+    </div>
+  );
+}
+
+export type Challenges = NonNullable<Awaited<ReturnType<typeof getChallenges>>>;
+async function getChallenges() {
+  return prisma.challenge.findMany({
+    where: {
+      slug: {
+        in: slugs,
+      },
+    },
+  });
+}
+
+const MS_PER_DAY = 24 * 60 * 60 * 1000;
+function revealItems(items: Challenges) {
```

Comment:
👍

Reply:
thought: This looks like it runs on server and uses server time. Looks great! I had my eye out to be careful we don't use local browser time so someone could set their date/time in the future and get all the challenges.

Reply:
ya know I didnt even consider that. that being said it will all be eastern time (US EAST 1). so we'll have to make the announcement with timezone in mind

Reply:
midnight ET, it's like staying up to watch the New York new years eve celebration ⭐